### PR TITLE
Identify unit tests that leave temporary data behind and add clean up

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/golemfactory/golem/issues/3890.

### CODE REVIEW

File: tests/apps/blender/task/test_blenderrendertask.py [hunk 1]
```diff
@@ -634,6 +634,15 @@ def test_error_in_preview_update(self):
 
 class TestBlenderRenderTaskBuilder(TempDirFixture):
 
+    def setUp(self):
+        super().setUp()
+        _, self.blender_file_path = tempfile.mkstemp('.blend')
```
Comment (Krigpl) [inline tests/apps/blender/task/test_blenderrendertask.py:639]: that could just use `self.tmp_dir` and get cleaned automatically

File: tests/apps/blender/benchmark/test_blenderbenchmark.py [hunk 2]
```diff
@@ -26,6 +26,11 @@ class TestBlenderBenchmark(unittest.TestCase, testutils.PEP8MixIn):
 
     def setUp(self):
         self.bb = BlenderBenchmark()
+        self.addCleanup(self.__clean_files)
+
+    def __clean_files(self):
+        if os.path.isfile(self.bb.task_definition.output_file):
+            os.remove(self.bb.task_definition.output_file)
```
Comment (Krigpl) [inline tests/apps/blender/benchmark/test_blenderbenchmark.py:33]: that doesn't feel like the right solution here - in production code we still leave this temp file.  imo benchmark classes should clean up after themselves

File: tests/apps/dummy/benchmark/test_dummybenchmark.py [hunk 3]
```diff
@@ -11,6 +12,11 @@ class TestDummyBenchmark(TempDirFixture):
     def setUp(self):
         super().setUp()
         self.db = DummyTaskBenchmark()
+        self.addCleanup(self.__clean_files)
+
+    def __clean_files(self):
+        if os.path.isdir(self.db._task_definition.tmp_dir):
+            shutil.rmtree(self.db._task_definition.tmp_dir)
```
Comment (Krigpl) [inline tests/apps/dummy/benchmark/test_dummybenchmark.py:19]: same as above

File: tests/golem/docker/test_docker_job.py [hunk 4]
```diff
@@ -37,11 +37,11 @@ def _get_test_tag(self):
     TEST_SCRIPT = "print 'Adventure Time!'\n"
 
     def setUp(self):
-        main_dir = get_local_datadir('tests-' + str(uuid.uuid4()))
-        if not os.path.exists(main_dir):
-            os.makedirs(main_dir)
+        self.main_dir = get_local_datadir('tests-' + str(uuid.uuid4()))
```
Comment (Krigpl) [inline tests/golem/docker/test_docker_job.py:40]: Why not just use `self.tmp_dir` here and below ?

**Review by Krigpl — COMMENTED** [review]
That looks like the majority of the leftover files comes from the production code that doesn't clean them up rather than anything specific to tests - so it'd be better to look into how to improve production code.

**Comment by rwrzesien** [comment]
@Krigpl 
> That looks like the majority of the leftover files comes from the production code that doesn't clean them up rather than anything specific to tests - so it'd be better to look into how to improve production code.

Probably yes, but a separate issue should be created for that. This one is specifically about cleaning after what unit tests run leaves.

File: tests/apps/dummy/benchmark/test_dummybenchmark.py (hunk 3 above)
Comment (rwrzesien) [reply tests/apps/dummy/benchmark/test_dummybenchmark.py:19]: See https://github.com/golemfactory/golem/pull/3920#issuecomment-466386794.

File: tests/apps/blender/benchmark/test_blenderbenchmark.py (hunk 2 above)
Comment (rwrzesien) [reply tests/apps/blender/benchmark/test_blenderbenchmark.py:33]: See https://github.com/golemfactory/golem/pull/3920#issuecomment-466386794.

File: tests/apps/blender/task/test_blenderrendertask.py (hunk 1 above)
Comment (rwrzesien) [reply tests/apps/blender/task/test_blenderrendertask.py:639]: I don't want to modify any existing tests that I am not aware what they are doing and what they suppose to really test.

File: tests/golem/docker/test_docker_job.py (hunk 4 above)
Comment (rwrzesien) [reply tests/golem/docker/test_docker_job.py:40]: I don't want to modify any existing tests that I am not aware what they are doing and what they suppose to really test.

**Review by Krigpl — CHANGES_REQUESTED** [review]
These tests expose an issue with production code and I'm not fine with just applying some bandaid solution to improve tests while sweeping the real problem under the rug.
Please modify `test_docker_job` and `test_blenderrendertask` as suggested, remove all the benchmark related hacks and create an issue to fix leftover benchmarks files.

**Comment by Krigpl** [comment]
Any progress on that?

**Comment by mplebanski** [comment]
Can we close this?

**Comment by lukasz-tarka** [comment]
If the solution for the problem explicitly described as test related in the issue is not acceptable, it should be closed and a new issue opened for fixing production code instead of tests itself.

**Comment by Krigpl** [comment]
rip [*]

**Pull request closed without merging**